test/unittest/xmlparser/XMLElementParserTests.cpp

```diff
@@ -307,6 +307,109 @@ TEST_F(XMLParserTests, get_xml_disable_heartbeat_piggyback)
+    char xml[xml_len];
+
+    // Valid XML
+    snprintf(xml, xml_len, xml_p, "8844", "192.168.1.55", "");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
+    EXPECT_EQ(list.begin()->port, 8844u);
+    EXPECT_EQ(list.begin()->address[12], 192);
+    EXPECT_EQ(list.begin()->address[13], 168);
+    EXPECT_EQ(list.begin()->address[14], 1);
+    EXPECT_EQ(list.begin()->address[15], 55);
+    EXPECT_EQ(list.begin()->kind, LOCATOR_KIND_UDPv4);
+
+    // Missing data - port
+    snprintf(xml, xml_len, xml_p, "", "192.168.1.55", "");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
+
+    // Missing data - address
+    snprintf(xml, xml_len, xml_p, "8844", "", "");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
+
+    // Invalid element
+    snprintf(xml, xml_len, xml_p, "8844", "192.168.1.55", "<bad_element></bad_element>");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
+}
+
+/*
+ * This test checks the parsing of a <udpv4> element from a list of locators, using DNS resolution.
+ * 1. Correct parsing of a valid element (with address given by domain).
+ */
+TEST_F(XMLParserTests, getXMLLocatorDNSUDPv4)
+{
+
+    uint8_t ident = 1;
+    LocatorList_t list;
+    tinyxml2::XMLDocument xml_doc;
+    tinyxml2::XMLElement* titleElement;
+
+    // Parametrized XML
+    const char* xml_p =
+            "\
+            <unicastLocatorList>\
+                <locator>\
+                    <udpv4>\
+                        <port>%s</port>\
+                        <address>%s</address>\
+                        %s\
+                    </udpv4>\
+                </locator>\
+            </unicastLocatorList>\
+            ";
+    constexpr size_t xml_len {500};
+    char xml[xml_len];
+
+    // Valid XML
+    snprintf(xml, xml_len, xml_p, "8844", "www.acme.com.test", "");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
+    EXPECT_EQ(list.begin()->port, 8844u);
+    EXPECT_EQ(list.begin()->address[12], 216);
+    EXPECT_EQ(list.begin()->address[13], 58);
+    EXPECT_EQ(list.begin()->address[14], 215);
+    EXPECT_EQ(list.begin()->address[15], 164);
+    EXPECT_EQ(list.begin()->kind, LOCATOR_KIND_UDPv4);
+}
```

Comment:
Could we add a failing test case to check that an invalid DNS name returns error? It applies to all 4 transports.